addon/components/polaris-data-table.js

```diff
@@ -0,0 +1,535 @@
+import Component from '@ember/component';
+import { computed } from '@ember/object';
+import { isBlank, isPresent, isNone } from '@ember/utils';
+import { htmlSafe } from '@ember/string';
+import { scheduleOnce, debounce } from '@ember/runloop';
+import { assign } from '@ember/polyfills';
+import layout from '../templates/components/polaris-data-table';
+
+const eventHandlerParams = [
+  [ window, 'resize', 'handleResize' ],
+  [ window, 'scroll', 'scrollListener', true ],
+];
+
+function volatileElementLookup(selector) {
+  return computed(function() {
+    return this.element.querySelector(selector);
+  }).volatile();
+}
+
+function measureColumn(tableData) {
+  return function(column, index) {
+    let {
+      tableLeftVisibleEdge,
+      tableRightVisibleEdge,
+      firstVisibleColumnIndex,
+      fixedColumnWidth,
+    } = tableData;
+    let width = column.offsetWidth;
+    let leftEdge = column.offsetLeft - fixedColumnWidth;
+    let rightEdge = leftEdge + width;
+    let leftEdgeIsVisible = isEdgeVisible(
+      leftEdge,
+      tableLeftVisibleEdge,
+      tableRightVisibleEdge,
+    );
+    let rightEdgeIsVisible = isEdgeVisible(
+      rightEdge,
+      tableLeftVisibleEdge,
+      tableRightVisibleEdge,
+    );
+    let isCompletelyVisible =
+      leftEdge < tableLeftVisibleEdge && rightEdge > tableRightVisibleEdge;
+    let isVisible =
+      isCompletelyVisible || leftEdgeIsVisible || rightEdgeIsVisible;
+    if (isVisible) {
+      tableData.firstVisibleColumnIndex = Math.min(
+        firstVisibleColumnIndex,
+        index,
+      );
+    }
+    return {leftEdge, rightEdge, isVisible};
+  };
+}
+
+function isEdgeVisible(target, start, end) {
+  let minVisiblePixels = 30;
+  return target >= start + minVisiblePixels && target <= end - minVisiblePixels;
+}
+
+function getPrevAndCurrentColumns(tableData, columnData) {
+  let {
+    tableRightVisibleEdge,
+    tableLeftVisibleEdge,
+    firstVisibleColumnIndex,
+  } = tableData;
+  let previousColumnIndex = Math.max(firstVisibleColumnIndex - 1, 0);
+  let previousColumn = columnData[previousColumnIndex];
+  let lastColumnIndex = columnData.length - 1;
+  let lastColumn = columnData[lastColumnIndex];
+  let currentColumn = assign(
+    {
+      isScrolledFarthestLeft:
+        firstVisibleColumnIndex === 0 && tableLeftVisibleEdge === 0,
+      isScrolledFarthestRight: lastColumn.rightEdge <= tableRightVisibleEdge,
+    },
+    columnData[firstVisibleColumnIndex]
+  );
+  return {previousColumn, currentColumn};
+}
+
+/**
+ * Polaris data table component.
+ * See https://polaris.shopify.com/components/lists-and-tables/data-table
+ */
+export default Component.extend({
+  classNameBindings: ['collapsed:Polaris-DataTable--collapsed'],
```

Comment:
Polaris adds a `TableWrapper` class here, I reckon there's no styles and not used...maybe we should still add it?

Reply:
Do you see that class being applied in actual usage? I checked this and nothing seems to end up getting applied to the element:

![image](https://user-images.githubusercontent.com/5737342/41973855-e91465a6-7a1e-11e8-9cf3-e1747d7afc14.png)

I guess there's no defined style for it in the Polaris source so it gets stripped out or something 🤷‍♂️

Reply:
yes most likely it's stripped out